inluxclient/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+// Package influxclient provides client for InfluxDB server.
+package influxclient
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+var (
+	// ErrEmptyServerURL is returned when params do have set ServerURL field
+	ErrEmptyServerURL = errors.New("empty server URL")
+)
+
+// Params holds the parameters for creating a new client.
+// The only mandatory field is ServerURL. AuthToken is also important
+// if authentication was not done outside this client.
+type Params struct {
+	// ServerURL holds the URL of the InfluxDB server to connect to.
+	// This must be non-empty. E.g. http://localhost:8086
+	ServerURL string
+
+	// AuthToken holds the authorization token for the API.
+	// This can be obtained through the GUI web browser interface.
+	AuthToken string
+
+	// DefaultTags specifies a set of tags that will be added to each written
+	// point. Tags specified on points override these.
+	DefaultTags map[string]string
+
+	// HTTPClient is used to make API requests.
+	//
+	// This can be used to specify a custom TLS configuration
+	// (TLSClientConfig), a custom request timeout (Timeout),
+	// or other customization as required.
+	//
+	// It HTTPClient is nil, http.DefaultClient will be used.
+	HTTPClient *http.Client
+
+	// BatchSize holds the default batch size
+	// used by PointWriter. If it's zero, DefaultBatchSize will
+	// be used. Note that this can be overridden with PointWriter.SetBatchSize.
+	BatchSize int
+
+	// FlushInterval holds the default flush interval used by PointWriter.
+	// If it's zero, points must be flushed manually.
+	// Note that this can be overridden with PointWriter.SetFlushInterval.
+	FlushInterval time.Duration
+}
+
+// Client implements an InfluxDB client.
+type Client struct {
+	// Configuration params.
+	params Params
+	// Pre-created Authorization HTTP header value.
+	authorization string
+	// Cached base server API URL.
+	apiURL *url.URL
```

Comment:
Given that you'll be needing to parse the URL again when making the request, I wonder if it might be best just to keep the URL in string form.

Reply:
The idea here is to avoid repeatedly parsing the complete URL string to the URL object, which will be used in specific API calls to construct endpoint URL with query params.